backend/authapp/views.py

```diff
@@ -0,0 +1,68 @@
+from rest_framework import status
+from rest_framework.generics import RetrieveUpdateAPIView
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .serializers import RegistrationSerializer, LoginSerializer, UserSerializer
+from .renderers import UserJSONRenderer
+
+
+class RegistrationAPIView(APIView):
+    # Allow any user (authenticated or not) to hit this endpoint.
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = RegistrationSerializer
+
+    def post(self, request):
+        user = request.data.get('user-registration', {})
+        print(user)
+
+        # The create serializer, validate serializer, save serializer pattern
+        # below is common and you will see it a lot throughout this course and
+        # your own work later on. Get familiar with it.
+        serializer = self.serializer_class(data=user)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        # return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(user)
+    
+    def get(self, request):
+        user = request.data.get('user')
+
+        return Response('Hi, Daniel')
```

Comment:
I think this was for testing purposes, if not used delete the whole get response